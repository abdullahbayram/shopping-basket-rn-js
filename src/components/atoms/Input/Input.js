import * as React from 'react';
import PropTypes from 'prop-types';
import { TextInput } from 'react-native-paper';
import { StyleSheet } from 'react-native';

const Input = ({ label, icon = null, onChangeText, maxLength, style, value, onBlur, onEndEditing }) => {
  return (
    <TextInput
      maxLength={maxLength}
      style={[styles.container, style]}
      label={label}
      onChangeText={onChangeText}
      right={<TextInput.Icon icon={icon || ''} />}
      value={String(value)}
      onBlur={onBlur}
      onEndEditing={onEndEditing}
    />
  );
};

Input.propTypes = {
  label: PropTypes.string,
  onChangeText: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  onEndEditing: PropTypes.func,
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  maxLength: PropTypes.oneOfType([PropTypes.number]),
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default Input;

const styles = StyleSheet.create({
  container: {
    marginVertical: 8,
  },
});
