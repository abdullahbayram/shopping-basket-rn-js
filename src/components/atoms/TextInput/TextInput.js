import { TextInput as PaperInput } from 'react-native-paper';

export default PaperInput;
