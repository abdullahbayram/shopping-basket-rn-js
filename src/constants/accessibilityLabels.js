const accessibilityLabels = {
  search: 'Search Button',
  more: 'More Options Button',
  back: 'Back Button',
};

export default accessibilityLabels;
