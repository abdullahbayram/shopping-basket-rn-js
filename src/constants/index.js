export { default as strings } from './strings';
export { default as toastMessages } from './toastMessages';
export { default as urls } from './urls';
