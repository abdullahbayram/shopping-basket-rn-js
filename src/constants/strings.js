export default {
  icons: {
    cardholderName: 'account',
    cardholderNameCheck: 'account-check',
    expirationDate: 'calendar-alert',
    expirationDateCheck: 'calendar-check',
    cvv: 'shield-alert',
    cvvCheck: 'shield-check',
    creditCardNumber: 'credit-card',
    creditCardNumberCheck: 'credit-card-check',
  },
  screens: {
    success: 'Success',
    payment: 'Payment',
    error: 'Error',
    checkout: 'Checkout',
  },
  devErrors: {
    parseErrorMessage: 'Failed to parse error message:',
  },
  buttons: {
    retry: 'Retry',
    checkout: 'CHECKOUT',
    order: 'ORDER',
    payAndOrder: 'PAY AND ORDER',
    gotoProducts: 'Go to Products',
  },
  productList: {
    basketItemCount: 'Items in the basket: ',
    loading: 'Loading products...',
    errorLoading: 'Unable to load products. Please try again.',
    limitReached: 'You reached the max quantity per product!',
  },
  checkout: {
    total: 'Total:',
    order: 'ORDER',
    promoCodePlaceholder: 'Enter your promo code',
    promoCode: 'Promo Code',
    promoCodeRequired: 'Promo code can not be empty',
    promoCodeNotValid: 'This promo code is not valid',
    applyPromo: 'APPLY PROMO CODE',
    emptyBasket: 'Your basket is empty.',
  },
  payment: {
    total: 'Total:',
    basketItemCount: 'Items in the basket: ',
    cardholderName: 'Cardholder Name',
    cardholderNamePlaceholder: 'Enter the name on your card',
    creditCardNumber: 'Credit Card Number',
    creditCardPlaceholder: 'Enter your credit card number',
    expirationDate: 'Expiration Date',
    expirationDatePlaceholder: 'MM/YY',
    cvv: 'CVV',
    cvvPlaceholder: '123',
    unexpectedError: 'An unexpected error occurred. Please try again later.',
    unexpectedErrorShort: 'Unexpected Error',
    invalidCard: 'Invalid credit card number',
    cardHolderRequired: 'Cardholder name is required',
    creditCardRequired: 'Credit card number is required',
    expirationDateRequired: 'Expiration date is required',
    cvvRequired: 'CVV is required',
    invalidExpirationDate: 'Invalid expiration date (MM/YY)',
    cvvLength: 'CVV must be 3 digits',
    cardHolderMinLength: 'Cardholder name must be at least 3 characters',
    success: `
Thank you!
Your order has been placed successfully.
      `,
  },
};
