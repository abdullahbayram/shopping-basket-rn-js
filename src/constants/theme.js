import { MD3LightTheme, MD3DarkTheme } from 'react-native-paper';

export const sharedColors = {
  overlayBackgroundColor: 'rgba(0,0,0,0.5)',
};

const lightColors = {
  primary: '#3498db',
  secondary: '#FED034',
  tertiary: '#34300F',
  error: '#FF6B6B',
  background: '#FFF',
  surface: '#F9F9F9',
  textPrimary: '#004f6b',
  textSecondary: '#666',
  textTertiary: '#000',
  linearLeft: '#6fc8dc',
  linearRight: '#8bd4c7',
  containerBackground: '#F9F9F9',
  shadowColor: 'rgba(0, 0, 0, 0.3)',
  buttonBackground: '#FFF',
  buttonBorder: '#FFA500',
  transparentButtonBorder: '#FFA500',
  cardBackground: '#F0F0F0',
  ratingStar: '#FFA500',
  textFour: '#333',
  disabledButton: '#ddd',
  disabledButtonBorder: '#aaa',
  get spinner() {
    return this.secondary;
  },
  progressBarBackground: '#e0e0e0',
  progressBar: '#76c7c0',
};

const darkColors = {
  primary: '#2980b9',
  secondary: '#E0A800',
  tertiary: '#FFF',
  error: '#FF6B6B',
  background: '#121212',
  surface: '#1E1E1E',
  textPrimary: '#E0E0E0',
  textSecondary: '#B3B3B3',
  textTertiary: '#FFF',
  linearLeft: '#2980b9',
  linearRight: '#4fb0c7',
  containerBackground: '#333',
  shadowColor: 'rgba(255, 255, 255, 0.1)',
  buttonBackground: '#444',
  buttonBorder: '#FFAA33',
  transparentButtonBorder: '#FFAA33',
  cardBackground: '#424242',
  ratingStar: '#FFD700',
  textFour: '#FFF',
  disabledButton: '#555',
  disabledButtonBorder: '#777',
  get spinner() {
    return this.secondary;
  },
  progressBarBackground: '#444',
  progressBar: '#76c7c0',
};

export const lightTheme = {
  ...MD3LightTheme,
  roundness: 2,
  colors: {
    ...MD3LightTheme.colors,
    ...lightColors,
  },
};

export const darkTheme = {
  ...MD3DarkTheme,
  roundness: 2,
  colors: {
    ...MD3DarkTheme.colors,
    ...darkColors,
  },
};

export const spacing = {
  xxs: 2,
  xs: 4,
  sm: 10,
  md: 16,
  lg: 24,
  xl: 32,
  special: 7,
  baseScreenSpecial: 9,
};
