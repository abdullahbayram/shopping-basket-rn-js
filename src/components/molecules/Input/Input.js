import * as React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import TextInput from '../../atoms/TextInput/TextInput';

const Input = ({ label = null, onChangeText, maxLength, style, value, onBlur, onEndEditing, right }) => {
  return (
    <TextInput
      maxLength={maxLength}
      style={[styles.container, style]}
      label={label}
      onChangeText={onChangeText}
      value={String(value)}
      onBlur={onBlur}
      onEndEditing={onEndEditing}
      right={right}
    />
  );
};

Input.propTypes = {
  label: PropTypes.string,
  onChangeText: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  onEndEditing: PropTypes.func,
  right: PropTypes.node,
  maxLength: PropTypes.oneOfType([PropTypes.number]),
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default Input;

const styles = StyleSheet.create({
  container: {
    marginVertical: 8,
  },
});
