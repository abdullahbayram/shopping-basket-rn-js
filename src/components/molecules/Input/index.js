import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import TextInput from '../../atoms/TextInput';
import HelperText from '../../atoms/HelperText';
import styles from './Input.style';

const Input = ({ label, onChangeText, maxLength, style, value = '', onBlur, onEndEditing, right, errorObject }) => {
  return (
    <View style={styles.container}>
      <TextInput
        maxLength={maxLength}
        style={style}
        label={label}
        onChangeText={onChangeText}
        value={value !== null && value !== undefined ? String(value) : ''}
        onBlur={onBlur}
        onEndEditing={onEndEditing}
        right={right}
        error={!!errorObject}
      />
      {errorObject ? (
        <HelperText type="error" visible={!!errorObject}>
          {errorObject?.message}
        </HelperText>
      ) : (
        <View style={styles.invisibleHeight} />
      )}
    </View>
  );
};

Input.propTypes = {
  label: PropTypes.string,
  onChangeText: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  onEndEditing: PropTypes.func,
  right: PropTypes.node,
  errorObject: PropTypes.shape({
    message: PropTypes.string,
  }),
  maxLength: PropTypes.oneOfType([PropTypes.number]),
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default Input;
