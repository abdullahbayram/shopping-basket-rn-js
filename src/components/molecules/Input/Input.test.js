import React from 'react';
import { render, screen, fireEvent, userEvent, waitFor } from '@testing-library/react-native';
import { snapshotDiff } from 'snapshot-diff/build';
import TextInput from '../../atoms/TextInput';
import Input from '.';
import { lightColors } from '../../../constants/theme';

const mockOnChangeText = jest.fn();
const mockOnBlur = jest.fn();
const mockOnEndEditing = jest.fn();

const renderInput = ({ label = 'Default Label', icon = null, maxLength, value = '', errorObject = null, style }) => {
  return render(
    <Input
      label={label}
      icon={icon}
      maxLength={maxLength}
      value={value}
      onChangeText={mockOnChangeText}
      onBlur={mockOnBlur}
      onEndEditing={mockOnEndEditing}
      errorObject={errorObject}
      style={style}
    />,
  );
};

describe('<Input />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with label and value', () => {
    renderInput({ value: 'Test Value', label: 'Test Label' });

    expect(screen.getAllByText('Test Label')).toBeTruthy();
    expect(screen.getByDisplayValue('Test Value')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const { toJSON } = renderInput({ value: 'Snapshot Value', label: 'Snapshot Label' });
    expect(toJSON()).toMatchSnapshot();
  });

  it('match snapshot difference with and without an icon', () => {
    const rightIcon = <TextInput.Icon testID="right-icon" icon="percent" color={lightColors.error} />;
    const inputWithIcon = renderInput({ right: rightIcon }).toJSON();
    const inputWithoutIcon = renderInput({ right: undefined }).toJSON();

    expect(snapshotDiff(inputWithIcon, inputWithoutIcon)).toMatchSnapshot();
  });

  it('calls onChangeText when the input text changes', () => {
    renderInput({ label: 'Input Label' });

    const inputElement = screen.getByTestId('text-input-flat');
    fireEvent.changeText(inputElement, 'New Value');

    expect(mockOnChangeText).toHaveBeenCalledWith('New Value');
  });

  it('calls onBlur when the input loses focus', async () => {
    renderInput({ label: 'Input Label' });
    const user = userEvent.setup();
    const inputElement = screen.getByTestId('text-input-flat');
    await waitFor(() => user.type(inputElement));
    await waitFor(() => user.clear(inputElement));

    expect(mockOnBlur).toHaveBeenCalled();
  });

  it('calls onEndEditing when editing ends', () => {
    renderInput({ label: 'Input Label' });

    const inputElement = screen.getByTestId('text-input-flat');
    fireEvent(inputElement, 'onEndEditing');

    expect(mockOnEndEditing).toHaveBeenCalled();
  });

  it('renders error message when errorObject is provided', () => {
    const errorMessage = 'This is an error';
    renderInput({ errorObject: { message: errorMessage } });

    expect(screen.getByText(errorMessage)).toBeTruthy();
  });

  it('does not render error message when errorObject is null', () => {
    renderInput({});

    const errorElement = screen.queryByText(/This is an error/i);
    expect(errorElement).toBeNull();
  });

  it('limits input length when maxLength is provided', () => {
    renderInput({ maxLength: 10 });

    const inputElement = screen.getByTestId('text-input-flat');
    fireEvent.changeText(inputElement, '123456789012345');

    expect(mockOnChangeText).toHaveBeenCalledWith('123456789012345'); // maxLength of 10
  });
});

describe('<Input /> edge cases for value prop', () => {
  it('handles null value gracefully', () => {
    renderInput({ value: null });

    const inputElement = screen.getByTestId('text-input-flat');
    expect(inputElement.props.value).toBe(''); // Null should be converted to an empty string
  });

  it('handles undefined value gracefully', () => {
    renderInput({ value: undefined });

    const inputElement = screen.getByTestId('text-input-flat');
    expect(inputElement.props.value).toBe(''); // Undefined should be converted to an empty string
  });

  it('handles empty string value gracefully', () => {
    renderInput({ value: '' });

    const inputElement = screen.getByTestId('text-input-flat');
    expect(inputElement.props.value).toBe(''); // Empty string remains as is
  });

  it('handles numeric value gracefully', () => {
    renderInput({ value: 123 });

    const inputElement = screen.getByTestId('text-input-flat');
    expect(inputElement.props.value).toBe('123'); // Number should be converted to a string
  });
});
