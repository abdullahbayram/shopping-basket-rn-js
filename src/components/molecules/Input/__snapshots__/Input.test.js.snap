// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> match the snapshot without an icon 1`] = `
<View
  style={
    {
      "marginBottom": 5,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(231, 224, 236, 1)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {},
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(73, 69, 79, 1)",
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
      testID="text-input-underline"
    />
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexGrow": 1,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 56,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 4,
                },
              ],
              "width": 750,
              "zIndex": 3,
            }
          }
        >
          <View
            style={
              {
                "width": 97.33333333333333,
              }
            }
          >
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              onLayout={[Function]}
              onTextLayout={[Function]}
              style={
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "maxWidth": 97.33333333333333,
                  "opacity": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 30,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateY": -12,
                    },
                    {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
              testID="text-input-flat-label-active"
            >
              Credit Card
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              style={
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "maxWidth": 97.33333333333333,
                  "opacity": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 30,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateY": -12,
                    },
                    {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
              testID="text-input-flat-label-inactive"
            >
              Credit Card
            </Text>
          </View>
        </View>
      </View>
      <TextInput
        cursorColor="rgba(103, 80, 164, 1)"
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=" "
        placeholderTextColor="rgba(73, 69, 79, 1)"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          [
            {
              "flexGrow": 1,
              "margin": 0,
            },
            {},
            {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "minWidth": 65,
              "paddingLeft": 16,
              "paddingRight": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
            undefined,
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value="1234 5678 9012 3456"
      />
    </View>
  </View>
  <View
    style={
      {
        "height": 20,
      }
    }
  />
</View>
`;

exports[`<Input /> matches the snapshot with label and icon 1`] = `
<View
  style={
    {
      "marginBottom": 5,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(231, 224, 236, 1)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {},
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(73, 69, 79, 1)",
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
      testID="text-input-underline"
    />
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexGrow": 1,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 56,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 4,
                },
              ],
              "width": 750,
              "zIndex": 3,
            }
          }
        >
          <View
            style={
              {
                "width": 97.33333333333333,
              }
            }
          >
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              onLayout={[Function]}
              onTextLayout={[Function]}
              style={
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "maxWidth": 97.33333333333333,
                  "opacity": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 30,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateY": -12,
                    },
                    {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
              testID="text-input-flat-label-active"
            >
              Credit Card
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              style={
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "maxWidth": 97.33333333333333,
                  "opacity": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 30,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateY": -12,
                    },
                    {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
              testID="text-input-flat-label-inactive"
            >
              Credit Card
            </Text>
          </View>
        </View>
      </View>
      <TextInput
        cursorColor="rgba(103, 80, 164, 1)"
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=" "
        placeholderTextColor="rgba(73, 69, 79, 1)"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          [
            {
              "flexGrow": 1,
              "margin": 0,
            },
            {},
            {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "minWidth": 65,
              "paddingLeft": 16,
              "paddingRight": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
            undefined,
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value="1234 5678 9012 3456"
      />
    </View>
  </View>
  <View
    style={
      {
        "height": 20,
      }
    }
  />
</View>
`;
